repair/repair.cc

```diff
@@ -914,12 +919,21 @@ future<> tasks::user_requested_repair_task_impl::run() {
 
         for (auto shard : boost::irange(unsigned(0), smp::count)) {
             auto f = rs.container().invoke_on(shard, [keyspace, table_ids, id, ranges, hints_batchlog_flushed,
-                    data_centers = options.data_centers, hosts = options.hosts, ignore_nodes] (repair_service& local_repair) mutable {
-                local_repair.get_metrics().repair_total_ranges_sum += ranges.size();
-                auto ri = make_lw_shared<repair_info>(local_repair,
-                        std::move(keyspace), std::move(ranges), std::move(table_ids),
-                        id, std::move(data_centers), std::move(hosts), std::move(ignore_nodes), streaming::stream_reason::repair, node_ops_id{id.uuid.uuid()}, hints_batchlog_flushed);
-                return repair_ranges(ri);
+                    data_centers = options.data_centers, hosts = options.hosts, ignore_nodes] (repair_service& local_repair) mutable -> future<>{
+                lw_shared_ptr<repair_info> ri;
+                std::exception_ptr ex;
+                auto task_id = tasks::task_id{id.uuid.uuid()};
+                try {
+                    local_repair.get_metrics().repair_total_ranges_sum += ranges.size();
+                    ri = make_lw_shared<repair_info>(local_repair,
+                            std::move(keyspace), std::move(ranges), std::move(table_ids),
+                            id, std::move(data_centers), std::move(hosts), std::move(ignore_nodes), streaming::stream_reason::repair, node_ops_id{id.uuid.uuid()}, hints_batchlog_flushed);
+                } catch (...) {
+                    ex = std::current_exception();
+                }
+                auto task_impl_ptr = std::make_unique<tasks::shard_repair_task_impl>(local_repair._repair_module, task_id::create_null_id(), 0, keyspace, "", format("{}", streaming::stream_reason::repair), "", task_id, ri, ex);
+                auto task = co_await start_repair_task(std::move(task_impl_ptr), local_repair._repair_module, task_manager::parent_data{task_id, this_shard_id()});
```

Comment:
don't we need to get the parent shard in the calling shard, ahead of the invoke_on contiuation?
maybe we can keep it in the repair id structure so to have the sequence number and parent_data (uuid and owning shard)